rutil/Data.cxx

```diff
@@ -352,6 +352,16 @@ Data::Data(const string& str)
    initFromString(str.c_str(), (Data::size_type)str.size());
 }
 
+#if RESIP_CPP_STANDARD >= 201703L
```

Comment:
The custom definition can be avoided in favour of the ones available for this goal - __cplusplus
Take a look at https://en.cppreference.com/w/cpp/preprocessor/replace for details.

Reply:
I'm using VS2022, and it seems that resip doesn’t have the __cplusplus macro enabled by default in all projects. For more information, please refer to this article [https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170](https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170). To avoid adding /Zc:__cplusplus to every Visual Studio project within resip, I chose to define a custom RESIP_CPP_STANDARD macro in compat.hxx.

Reply:
From my understanding, Scott prefers using the pre-configured Visual Studio projects for Windows.

Reply:
If you use the GCC, CLANG, that RESIP_CPP_STANDARD marro actually is __cplusplues: https://github.com/resiprocate/resiprocate/pull/381/files#diff-7e40ff119bc1d70812a9d4d5452f335eae75b85fa577411dc376a18b82ddab15

Reply:
Unfortunately the CMake system isn't working on windows yet.  It was contributed by an author that is no longer around, and I haven't had time to try to get it working.  In the meantime the Visual Studio solution and project files are the only current solution for Visual Studio/Windows users.  If anyone has CMake working on windows, I'd love to get a PR in place.  :)

Reply:
Perhaps you can check both the __cplusplus macro OR'd with this new custom macro for VS users.